src/Data/Label/Derive.hs

```diff
@@ -327,6 +337,19 @@ constructorFields con =
             setEqs (Field a b c d) = Field a b c (first upd . second (eqs ++) $ d)
             upd (Context a b c) = Context a b (ForallC x y c)
 
+#if MIN_VERSION_template_haskell(2,11,0)
+    RecGadtC cs fs _ty -> concatMap (\c -> one c <$> zip [0..] fs) cs
```

Comment:
Should this `_ty` (the type constucted for this GADT constructor) be used instead of the other `ty` used in the body of the where clause?